framework/encode/d3d12_capture_manager.cpp

```diff
@@ -380,19 +383,32 @@ void D3D12CaptureManager::InitializeID3D12ResourceInfo(ID3D12Device_Wrapper*
     }
 }
 
-void D3D12CaptureManager::InitializeSwapChainBufferResourceInfo(ID3D12Resource_Wrapper* resource_wrapper,
+void D3D12CaptureManager::InitializeSwapChainBufferResourceInfo(IDXGISwapChain_Wrapper* wrapper,
+                                                                ID3D12Resource_Wrapper* resource_wrapper,
                                                                 D3D12_RESOURCE_STATES   initial_state)
 {
     GFXRECON_ASSERT(resource_wrapper != nullptr);
     GFXRECON_ASSERT(resource_wrapper->GetObjectInfo() != nullptr);
 
     auto info = resource_wrapper->GetObjectInfo();
+    GFXRECON_ASSERT(info);
+
+    graphics::dx12::ID3D12DeviceComPtr device;
+    DecrementCallScope();
+    wrapper->GetDevice(IID_PPV_ARGS(&device));
+    IncrementCallScope();
+    GFXRECON_ASSERT(device);
+
+    ID3D12Device_Wrapper* device_wrapper = nullptr;
+    device->QueryInterface(IID_IUnknown_Wrapper, reinterpret_cast<void**>(&device_wrapper));
+    GFXRECON_ASSERT(device_wrapper);
```

Comment:
Is it necessary to use `DecrementCallScope` here? `if (call_scope > 1)` then `ID3D12DeviceChild_Wrapper::GetDevice` returns `GetWrappedObjectAs<ID3D12DeviceChild>()->GetDevice(riid, ppvDevice);` which looks like it should be the native DX12 ID3D12Device, not a wrapped object. Then `ID3D12Device_Wrapper::GetExistingWrapper` could be used to get the device wrapper.

For example:
```
// Get the swapchain's native ID3D12Device
graphics::dx12::ID3D12DeviceComPtr device;
wrapper->GetDevice(IID_PPV_ARGS(&device)); 

// Get the ID3D12Device_Wrapper
ID3D12Device_Wrapper* device_wrapper = ID3D12Device_Wrapper::GetExistingWrapper(device);
```

Reply:
fixed